cmd/plugin/builder/inventory/inventory_plugin.go

```diff
@@ -137,10 +137,17 @@ func (ipuo *InventoryPluginUpdateOptions) updatePluginInventoryEntry(pluginInven
 	log.Infof("validating plugin '%s_%s_%s_%s'", plugin.Name, plugin.Target, osArch.String(), version)
 
 	pluginImageBasePath := fmt.Sprintf("%s/%s/%s/%s/%s/%s:%s", ipuo.Vendor, ipuo.Publisher, osArch.OS(), osArch.Arch(), plugin.Target, plugin.Name, version)
-	pluginImage := fmt.Sprintf("%s/%s", ipuo.Repository, pluginImageBasePath)
-	digest, err := ipuo.ImgpkgOptions.GetFileDigestFromImage(pluginImage, cli.MakeArtifactName(plugin.Name, osArch))
-	if err != nil && !ipuo.ValidateOnly {
-		return nil, errors.Wrapf(err, "error while getting plugin binary digest from the image %q", pluginImage)
+
+	var digest string
+	if !ipuo.ValidateOnly {
+		// If we are only validating the plugin's existence, we don't need to waste
+		// resources downloading the image to get the digest which won't actually be used.
```

Comment:
This is part of validation as we are actually validating that the plugins are available on the repository before we can add them to the database. So, we might need this check even for validation.

Reply:
What we can possibly improve is to check the image exists by just checking the container image digest and not downloading the entire image to improve this. But we do need to validate that image exists on the repository.

Reply:
This change disables the download of the image just when we use the `--validate` flag.
When the `--validate` flag is used, I believe we do **not** require the OCI image to be there for each plugin.
This check: https://github.com/vmware-tanzu/tanzu-cli/blob/9636a06b4d9efc64c38597b5d5c0fd48ff704153/cmd/plugin/builder/inventory/inventory_plugin.go#L142-L144
will ignore the missing OCI image `if ipuo.ValidateOnly == true`

Reply:
I think this makes sense. Thanks for explaining this.